File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-site-update.model.ts
import { HealthAuthoritySite } from '@health-auth/shared/models/health-authority-site.model';

export interface HealthAuthoritySiteUpdate extends Pick<HealthAuthoritySite,
  'siteName' |
  'pec' |
  'pharmacyPhone' |
  'mnemonic' |
  'securityGroupCode' |
  'physicalAddress' |
  'businessHours' |
  'healthAuthorityPharmanetAdministratorId' |
  'healthAuthorityTechnicalSupportId'> {
  healthAuthorityVendorId: number;
  healthAuthorityCareTypeId: number;
}

File: prime-dotnet-webapi/ViewModels/Health Authority Sites/HealthAuthoritySiteUpdateModel.cs
using System.Collections.Generic;
using Prime.ViewModels.Sites;

namespace Prime.ViewModels.HealthAuthoritySites
{
    public class HealthAuthoritySiteUpdateModel
    {
        public string SiteName { get; set; }
        public string PEC { get; set; }
        public int? SecurityGroupCode { get; set; }
        public AddressViewModel PhysicalAddress { get; set; }
        public int HealthAuthorityVendorId { get; set; }
        public int? HealthAuthorityCareTypeId { get; set; }
        public int? HealthAuthorityPharmanetAdministratorId { get; set; }
        public int? HealthAuthorityTechnicalSupportId { get; set; }
        public ICollection<BusinessDayViewModel> BusinessHours { get; set; }
    }
}
